orca-keel/src/test/java/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTaskTest.java

```diff
@@ -0,0 +1,385 @@
+
+  private static final int keelPort = 8087;
+
+  @BeforeAll
+  static void setupOnce(WireMockRuntimeInfo wmRuntimeInfo) {
+    OkClient okClient = new OkClient();
+    RestAdapter.LogLevel retrofitLogLevel = RestAdapter.LogLevel.NONE;
+
+    keelService =
+        new RestAdapter.Builder()
+            .setRequestInterceptor(
+                new SpinnakerRequestInterceptor(new OkHttpClientConfigurationProperties()))
+            .setEndpoint(wmRuntimeInfo.getHttpBaseUrl())
+            .setClient(okClient)
+            .setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
+            .setLogLevel(retrofitLogLevel)
+            .setConverter(new JacksonConverter(objectMapper))
+            .build()
+            .create(KeelService.class);
+
+    scmService = mock(ScmService.class);
+  }
+
+  @BeforeEach
+  public void setup() {
+    importDeliveryConfigTask = new ImportDeliveryConfigTask(keelService, scmService, objectMapper);
+
+    PipelineExecutionImpl pipeline = new PipelineExecutionImpl(PIPELINE, "keeldemo");
+    contextMap.put("repoType", "stash");
+    contextMap.put("projectKey", "SPKR");
+    contextMap.put("repositorySlug", "keeldemo");
+    contextMap.put("directory", ".");
+    contextMap.put("manifest", "spinnaker.yml");
+    contextMap.put("ref", "refs/heads/master");
+    contextMap.put("attempt", 1);
+    contextMap.put("maxRetries", 5);
+
+    stage = new StageExecutionImpl(pipeline, ExecutionType.PIPELINE.toString(), contextMap);
+  }
+
+  @RegisterExtension
+  static WireMockExtension wireMock =
+      WireMockExtension.newInstance().options(wireMockConfig().port(keelPort)).build();
+
+  private static void simulateFault(String url, String body, HttpStatus httpStatus) {
+    wireMock.givenThat(
+        WireMock.post(urlPathEqualTo(url))
+            .willReturn(aResponse().withStatus(httpStatus.value()).withBody(body)));
+  }
+
+  private static void simulateFault(String url, Fault fault) {
+    wireMock.givenThat(WireMock.post(urlPathEqualTo(url)).willReturn(aResponse().withFault(fault)));
+  }
+
+  /**
+   * This test is a positive case which verifies if the task returns {@link
+   * ImportDeliveryConfigTask.SpringHttpError} on 4xx http error. Here the error body is mocked with
+   * timestamp in {@link ChronoUnit#MILLIS}, which will be parsed to exact same timestamp in the
+   * method @see {@link ImportDeliveryConfigTask#handleRetryableFailures(RetrofitError,
+   * ImportDeliveryConfigTask.ImportDeliveryConfigContext)} and results in successful assertions of
+   * all the fields.
+   *
+   * <p>Other positive scenarios when the timestamp results in accurate value, are when the units in
+   * : {@link ChronoUnit#SECONDS} {@link ChronoUnit#DAYS} {@link ChronoUnit#HOURS} {@link
+   * ChronoUnit#HALF_DAYS} {@link ChronoUnit#MINUTES}
+   */
+  @Test
+  public void verifyTaskReturnsSpringHttpError() throws JsonProcessingException {
+
+    var mockResponseBody =
```

Comment:
```suggestion
    var scmResponseBody =
```

Reply:
This part is extracted to a private method : getDeliveryConfigManifest() and the response body is directly injected to the expected response instead of assigning to a variable.